homeassistant/components/light/mqtt_json.py

```diff
@@ -180,11 +202,13 @@ def turn_on(self, **kwargs):
         message = {'state': 'ON'}
 
         if ATTR_RGB_COLOR in kwargs:
-            message['color'] = {
-                'r': kwargs[ATTR_RGB_COLOR][0],
-                'g': kwargs[ATTR_RGB_COLOR][1],
-                'b': kwargs[ATTR_RGB_COLOR][2]
-            }
+            message.update(json.loads(self._templates[CONF_RGB_SET_TEMPLATE](
```

Comment:
Controlling the JSON that is sent is not crazy IMO. I don't see how that is confusing for the end user as the end user of this component has the skill to craft his own light and program it himself. 
The targeted audience of this component is already developers/makers and this PR just gives them more control over reading/writing the above mentioned JSON.